### PR TITLE
examples/shaders: Add an example for deferred shading

### PR DESCRIPTION
**Pull request by 27justin**
Adds an example for deferred shading using a G-Buffer and a rudimentary lighting shader operating on the G-Buffer.

### CODE REVIEW

File: examples/shaders/resources/shaders/glsl330/deferred_shading.fs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+in vec2 texCoord;
+in vec2 texCoord2;
+
+uniform sampler2D gPosition;
+uniform sampler2D gNormal;
+uniform sampler2D gAlbedoSpec;
+
+struct Light {
+	int enabled;
```
Comment (raysan5) [inline examples/shaders/resources/shaders/glsl330/deferred_shading.fs:12]: Please, could you replace TABS by 4-spaces?

File: examples/shaders/shaders_deferred_shading.c [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+    Shader deferredShader = LoadShader("resources/shaders/glsl330/deferred_shading.vs",
+                               "resources/shaders/glsl330/deferred_shading.fs");
+    deferredShader.locs[SHADER_LOC_VECTOR_VIEW] = GetShaderLocation(deferredShader, "viewPosition");
+
+
+    // Initialize the G-buffer
+    GBuffer gBuffer = { 0 };
+    gBuffer.framebuffer = rlLoadFramebuffer(screenWidth, screenHeight);
+
+    if(!gBuffer.framebuffer) {
```
Comment (raysan5) [inline examples/shaders/shaders_deferred_shading.c:78]: Please, could you review raylib coding conventions to use aligned `{` and one space after `if`/`for`/`switch`...

**Comment by raysan5** [comment]
@27justin Thank you very much for this great example! Please, could you review the mentioned points?

**Comment by 27justin** [comment]
Hey @raysan5, I amended the sources as per your comments and the project convention. 
Let me know if there's more to fix :+1:

**Comment by raysan5** [comment]
@27justin that's perfect! Thank you very much for such a nice addition! 😄

**Pull request merged**